Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -2062,6 +2062,58 @@ class SwiftMongoClientTests: SwiftSyncTestCase {
     }
 }
 
+class AnyRealmValueSyncTests: SwiftSyncTestCase {
+    /// The purpose of this test is to confirm that when an Object is set on a mixed Column and an old
+    /// version of an app does not have that Realm Object / Schema, we must fail gracefully and return nil
+    /// when calling `obj!.anyCol.value.object(Object.self)`
```

Comment:
We probably want to be returning a dynamic object with a dynamically created ObjectSchema unless that's impossible to implement.

Reply:
Creating an object dynamically is fine. Getting the RLMDynamicObject to cast to a Swift DynamicObject, or even RealmSwiftObject is an issue.

Here is how I'm creating the object: https://github.com/realm/realm-cocoa/blob/1a1c0c7e7f95b7bbb2be7983ee8a8f36d7bb3f30/Realm/RLMObjectStore.mm#L183

Reply:
The only thing I can think of to get around this and add the proper swift castings is to have a property on the realm instance itself that would indicate that the access came from a swift caller.

Reply:
I have added https://github.com/realm/realm-cocoa/blob/c5fb724db3f3a1bc83e7ee5ba3ba14f0ca131dfc/RealmSwift/AnyRealmValue.swift#L148